Update Makefile: use XcodeGen instead of SPM to create .xcodeproj

zoul opened the pull request:
In #18 I forgot to update the Makefile, so that running `make xcodeproj` would still use SPM to generate the Xcode project file. Not what we want.

zoul committed:
Update Makefile: use XcodeGen instead of SPM to create .xcodeproj

In #18 I forgot to update the Makefile, so that running `make xcodeproj`
would still use SPM to generate the Xcode project file. Not what we want.

zoul commented:
I see, the `xcodeproj` target is used as a dependency for `test-swift`, and that makes it fail on Linux. That looks as a simple omission, as the Xcode project file should not be needed on Linux?

stephencelis commented:
Ah, thanks so much for the catch!

stephencelis commented:
Ah yup, `test-linux` should definitely not depend on `xcodeproj`. Wanna delete that and I can merge?

zoul committed:
Remove incorrect Makefile dependency

The test-swift target was incorrectly depending on xcodeproj, breaking the Linux build.

mbrandonw commented:
Awesome, thanks for fixing this!
